guides/how_sprockets_works.md

```diff
@@ -38,132 +40,135 @@ Sprockets has some key components that are:
 - `:stubbed` - A Set of String asset URIs that will be omitted from the `:required` set
 - `:links` - A Set of String asset URIs that should be compiled along with the assets
 - `:dependencies` - A Set of String cache URIs that should be monitored for caching
 - `:map` - An Array of source maps for the assets
-- `:charset` - The mime charset for an asset
+- `:charset` - The MIME charset for an asset
 
- The `:required` is really interesting(as we will see later), each dependency from your asset files will be stored in this field.
+As we will see later, the `:required` is really interesting. Each dependency from your asset files will be stored in this field.
 
-There are a lot of interesting built-in processors as:
+There are a lot of interesting built-in processors including:
 
-- BabelProcessor
-- CoffeScriptProcessor
-- SassProcessor
-- BundlerProcessor etc...
+- `BabelProcessor`
+- `CoffeScriptProcessor`
+- `SassProcessor`
+- `BundlerProcessor`
 
- BundlerProcessor is used to run concatenated assets rather than individual files.
+`BundlerProcessor` is used to run concatenated assets rather than individual files.
 
-To register a processor in Sprockets, we use this syntax.
+To register a processor in Sprockets, we use this syntax:
 
 ```ruby
 register_bundle_processor 'application/javascript', Bundle
 register_bundle_processor 'text/css', Bundle
 ```
 
- We are telling that for any `application/javascript` mime type file, we are using the `Bundle` processor to take care of these files and concatenating them in the same file. So the `Bundle` processor takes a single file asset and prepends all the `required` URIs in the contents.
- 
+We are saying that for any file with MIME type `application/javascript`, we are using the `Bundle` processor to take care of these files and concatenating them in the same file. So the `Bundle` processor takes a single file asset and prepends all the `required` URIs in the contents.
+
 ### Transformers
 
-A transformer is a processor that converts a file from one format to another format. So, one of the examples is the CoffeeScript transformer that gets CoffeeScript file and returns a JavaScript file.
+A transformer is a processor that converts a file from one format to another format. One of the examples is the `CoffeeScript` transformer that takes a `CoffeeScript` file and returns a JavaScript file.
 
 ```ruby
 register_transformer 'text/coffescript', 'application/javascript', CoffeScriptProcessor
 ```
 
-The permutation of these processors are really simple.
+The implementation of these processors is really simple as we can see below:
 
 ```ruby
 module CoffeScriptProcessor
-  #......
 
   def self.call(input)
     data = input[:data]
 
      js, map = input[:cache].fetch([self.cache_key, data]) do
        result = CoffeScript.compile(data, sourceMap: true, sourceFiles: [input[:source_path]])
-       [result[‘js’], decode_source_maps(result[‘v3SourceMap’])]
```

Comment:
The reasoning behind changing from typographic quotes to simple quotes is that it makes copying and pasting from the example code less painful.